pkg/query/selection.go

```diff
@@ -195,16 +197,26 @@ func BuildCriteriaFromRequest(request *web.Request) ([]Criterion, error) {
 	return criteria, nil
 }
 
+var (
+	escapedOpenBracket      = url.QueryEscape(OpenBracket)
+	escapedCloseBracket     = url.QueryEscape(CloseBracket)
+	escapedSeparator        = url.QueryEscape(string(Separator))
+	escapedOperandSeparator = url.QueryEscape(OperandSeparator)
+	escapeReplacer          = strings.NewReplacer(escapedOpenBracket, OpenBracket, escapedCloseBracket, CloseBracket, escapedSeparator, string(Separator), escapedOperandSeparator, OperandSeparator, "%3D", "=")
+)
+
 func process(input string, criteriaType CriterionType) ([]Criterion, error) {
 	var c []Criterion
 	if input == "" {
 		return c, nil
 	}
+	escapedInput := url.QueryEscape(input)
+	escapedInput = escapeReplacer.Replace(escapedInput)
```

Comment:
not sure if this works as initally desired. This look especially confusing when for example one tries querying say service_plans by description that equals Shared fake Server, 5tb persistent disk, 40 max concurrent connections. 100 async.  which would be the similar to the example description in the osb spec